packages/wonder-blocks-core/src/util/add-style.tsx

```diff
@@ -5,6 +5,40 @@ import {processStyleList} from "./util";
 
 import type {StyleType} from "./types";
 
+type StyledProps<T extends keyof JSX.IntrinsicElements> = {
+    className?: string;
+    style?: StyleType;
+    children?: React.ReactNode;
+} & Omit<React.ComponentProps<T>, "style">;
+
+interface StyledTag<T extends keyof JSX.IntrinsicElements> {
```

Comment:
Some mind-bending typing here! Can you add code comments to help consumers understand what it does?

Reply:
I commente the function itself below since that's the only exported part -- these types wouldn't have any other consumers, but I can certainly comment for other devs' edification!

Reply:
That's a good point! I suppose the workflow I'm thinking of is when devs dig in more to understand what the code does. Thanks @nedredmond!

Reply:
@marcysutton Do you think I should prepare this for a real merge? I made it a draft as an illustration of a solution but I'll add a `changeset` if you'd like to pull it into wonder-blocks. It _will_ mark `addStyle` as deprecated!